### ISSUE TITLE
Index value is omitted on request payload, if index value is 0.

### ISSUE THREAD

**Issue by NikunjPansuriya** [piusalfred/whatsapp#31]
I am passing button with index 0 in template message but in request payload its omitted.
```
buttonParams := []*models.TemplateParameter{{
	Type: "text",
	Text: "1234",
}}
components := []*models.TemplateComponent{
	{
	    Type:       "button",
	    SubType:    "url",
	    Index:      0,
	    Parameters: buttonParams,
	},
}
```

Generated payload
```
{"messaging_product":"whatsapp","to":"+***********","recipient_type":"individual","type":"template","template":{"name":"templ1","language":{"code":"en_US"},"components":[{"type":"button","sub_type":"url","parameters":[{"type":"text","text":"1234"}]}]}}

I debugged issue and found that its because of `omitempty` defined in TemplateComponent struct.

**Comment by piusalfred** [comment]
Nice catch. I admit that I did not test this as much as I should, I will try to fix this if I get time together with #30. Feel Free to submit a fix though if you are blocked.

**Comment by piusalfred** [comment]
Wondering if changing all the fields that are optional to pointers. For now I will just remove the `omitempty`
